src/promptflow/promptflow/azure/operations/_arm_connection_operations.py

```diff
@@ -0,0 +1,256 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+from enum import Enum
+from typing import Any, Dict, Union
+
+import requests
+from azure.ai.ml._restclient.v2023_06_01_preview.models import WorkspaceConnectionPropertiesV2BasicResource
+from azure.ai.ml._scope_dependent_operations import (
+    OperationConfig,
+    OperationsContainer,
+    OperationScope,
+    _ScopeDependentOperations,
+)
+
+from promptflow._sdk.entities._connection import CustomConnection, _Connection
+from promptflow.azure._restclient.flow_service_caller import FlowServiceCaller
+from promptflow.exceptions import ErrorTarget, SystemErrorException, UserErrorException
+
+GET_CONNECTION_URL = (
+    "/subscriptions/{sub}/resourcegroups/{rg}/providers/Microsoft.MachineLearningServices"
+    "/workspaces/{ws}/connections/{name}/listsecrets?api-version=2023-04-01-preview"
+)
+LIST_CONNECTION_URL = (
+    "/subscriptions/{sub}/resourcegroups/{rg}/providers/Microsoft.MachineLearningServices"
+    "/workspaces/{ws}/connections?api-version=2023-04-01-preview"
+)
+FLOW_META_PREFIX = "azureml.flow."
+
+
+class ConnectionCategory(str, Enum):
+    AzureOpenAI = "AzureOpenAI"
+    CognitiveSearch = "CognitiveSearch"
+    CognitiveService = "CognitiveService"
+    CustomKeys = "CustomKeys"
+
+
+class ArmConnectionOperations(_ScopeDependentOperations):
+    """ArmConnectionOperations.
+
+    Get connections from arm api. You should not instantiate this class directly. Instead, you should
+    create an PFClient instance that instantiates it for you and
+    attaches it as an attribute.
```

Comment:
Add param list docstring?

Reply:
Will refine it in the future, as this is a private class inherited from azure ai, it will not be seen by customers.